### PR TITLE
SDKS-2989: My segments v2 Bounded (streaming optimization)-

### PR DESCRIPTION
**Pull request by javrudsky**
# iOS SDK

## What did you accomplish?
- Added logic to handle my segments bounded notification and tests 
- Updated buffer size in zlib decompression

### CODE REVIEW

File: Split/Network/Streaming/SyncUpdateWorker.swift [hunk 1]
```diff
@@ -146,11 +161,20 @@ class MySegmentsUpdateV2Worker: UpdateWorker<MySegmentsUpdateV2Notification> {
...
-        if keyList.removed.contains(hashedKey) {
+        if keyList.removed.contains(keyHash) {
             remove(segment: segmentName)
             return
         }
     }
+
+    private func handleBounded(encodedKeyMap: String, compressionUtil: CompressionUtil) throws {
+        let keyMap = try payloadDecoder.decodeAsBytes(payload: encodedKeyMap, compressionUtil: compressionUtil)
+        let index = payloadDecoder.computeKeyIndex(hashedKey: keyHash, keyMapLength: keyMap.count)
```
Comment (EmilianoSanchez) [inline Split/Network/Streaming/SyncUpdateWorker.swift:172]: if both `computeKeyIndex` and `isKeyInBitmap` methods are always used together, we can consider moving `computeKeyIndex` logic inside `isKeyInBitmap`.

Comment (EmilianoSanchez) [reply Split/Network/Streaming/SyncUpdateWorker.swift:172]: `isKeyInBitmap` signature might be like  `func isKeyInBitmap(keyMap: Data, hashedKey: UInt64) -> Bool`

File: Split/Network/Streaming/SyncUpdateWorker.swift [hunk 2]
```diff
@@ -100,25 +100,44 @@ class MySegmentsUpdateV2Worker: UpdateWorker<MySegmentsUpdateV2Notification> {
...
+                                       segmentName: segmentName,
+                                       compressionUtil: decompressor(for: notification.compressionType))
+                }
+            case .segmentRemoval:
+                if let segmentName = notification.segmentName {
+                    remove(segment: segmentName)
+                }
+            case .unknown:
+                // should never reach here
+                print("Unknown my segment v2 update strategy received")
```
Comment (EmilianoSanchez) [inline Split/Network/Streaming/SyncUpdateWorker.swift:124]: maybe `print` --> `Logger.i` ?

**Review by EmilianoSanchez — COMMENTED** [review]
Just a few comments to check Javi

File: Split/Network/Streaming/SyncUpdateWorker.swift (hunk 1 above)
Comment (javrudsky) [reply Split/Network/Streaming/SyncUpdateWorker.swift:172]: yeah, good catch doc! It makes sense :)

**Review by EmilianoSanchez — APPROVED** [review]

File: Split/Network/Streaming/SyncUpdateWorker.swift (hunk 2 above)
Comment (javrudsky) [reply Split/Network/Streaming/SyncUpdateWorker.swift:124]: 🤦‍♂️ yeah!

**Pull request merged**